lib/Sema/CSDiagnostics.cpp

```diff
@@ -670,6 +670,18 @@ bool GenericArgumentsMismatchFailure::diagnoseAsError() {
       break;
     }
 
+    case ConstraintLocator::TupleElement: {
+      if (auto *array = dyn_cast<ArrayExpr>(getRawAnchor()))
```

Comment:
Since `array` is not used, perhaps you could just use an `isa`?

Reply:
Yeah, make sense. Thanks!